File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.js
/**
 * @param {number} n
 * @return {boolean}
 */
var hasAlternatingBits = function(n) {
    let cur=0,pre=-1;
    while(n!=0){
        cur=n&1;
        if(cur==pre) return false;
        n>>>=1;
        pre=cur;
    }
    return true;
};

File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
      int a=n&1;
        n=n>>1;
        while(n>0){
            if((n&1)==a){
                return false;
            }
            a=n&1;
            n=n>>1;
        }
        return true;
    }
}